extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/ingest/AbstractSimpleFeatureIngestPlugin.java

```diff
@@ -66,14 +67,21 @@ public void setTypeNameProvider(
 		this.typeNameProvider = typeNameProvider;
 	}
 
+	public void setSimpOptionProvider(
+			final GeometrySimpOptionProvider geometryProvider ) {
+		this.simpOptionProvider = geometryProvider;
+	}
+
 	@Override
 	public byte[] toBinary() {
 		final byte[] filterBinary = filterOptionProvider.toBinary();
 		final byte[] typeNameBinary = typeNameProvider.toBinary();
-		final ByteBuffer buf = ByteBuffer.allocate(filterBinary.length + typeNameBinary.length + 4);
+		final byte[] simpBinary = simpOptionProvider.toBinary();
```

Comment:
fromBinary was unchanged (and it looks like the size of the binary needs to be serialized to be able to deserialize)